src/Microsoft.Restier.Core/Conventions/ConventionBasedEntitySetProcessor.cs

```diff
@@ -53,48 +56,130 @@ public Expression Process(QueryExpressionContext context)
+            var method = this.targetType.GetQualifiedMethod(methodName);
+            if (method == null || ! method.IsFamily)
             {
                 return null;
             }
 
-            var entitySet = dataSourceStubReference.Element as IEdmEntitySet;
-            if (entitySet == null)
+            var parameters = method.GetParameters();
+            if (parameters.Length != 1 ||
+                parameters[0].ParameterType != method.ReturnType)
             {
                 return null;
             }
 
-            var returnType = context.VisitedNode.Type
-                .FindGenericType(typeof(IQueryable<>));
-            var elementType = returnType.GetGenericArguments()[0];
-            var methodName = ConventionBasedChangeSetConstants.FilterMethodEntitySetFilter + entitySet.Name;
-            var method = this.targetType.GetQualifiedMethod(methodName);
-            if (method != null && method.IsFamily &&
-                method.ReturnType == returnType)
+            object apiBase = null;
+            if (!method.IsStatic)
+            {
+                apiBase = context.QueryContext.GetApiService<ApiBase>();
+                if (apiBase == null ||
+                    !this.targetType.IsInstanceOfType(apiBase))
+                {
+                    return null;
+                }
+            }
+
+            var returnType = context.VisitedNode.Type.FindGenericType(typeof(IQueryable<>));
+            var enumerableQueryPara = (object)context.VisitedNode;
+            Type elementType = null;
+
+            if (returnType == null)
             {
-                object target = null;
-                if (!method.IsStatic)
+                // This means append for properties model reference
+                var collType = context.VisitedNode.Type.FindGenericType(typeof (ICollection<>));
+                if (collType == null)
                 {
-                    target = context.QueryContext.GetApiService<ApiBase>();
-                    if (target == null ||
-                        !this.targetType.IsInstanceOfType(target))
-                    {
-                        return null;
-                    }
+                    return null;
                 }
 
-                var parameters = method.GetParameters();
-                if (parameters.Length == 1 &&
-                    parameters[0].ParameterType == returnType)
+                elementType = collType.GetGenericArguments()[0];
+                returnType = typeof(IQueryable<>).MakeGenericType(elementType);
+
+                enumerableQueryPara = Expression.Call(
+                    ExpressionHelperMethods.QueryableAsQueryableGeneric.MakeGenericMethod(elementType),
+                    context.VisitedNode);
+            }
+            else
+            {
+                elementType = returnType.GetGenericArguments()[0];
+            }
+            
+            var queryType = typeof(EnumerableQuery<>).MakeGenericType(elementType);
```

Comment:
better add a comment to describe the equivalent LINQ expression from line 146-184.